grails-app/services/org/broadinstitute/orsp/ConsentService.groovy

```diff
@@ -513,6 +514,26 @@ class ConsentService implements Status {
         }
     }
 
+    /**
+     * Find all consent groups associated to the specified consent collection links
+     *
+     * @param keys The consent group key and related consent collection link
+     * @return List of Consent Group that are related to the specified consent collection links
+     */
+    Collection<Issue> findConsentGroupsAssociatedByConsentLink(Map<String, ConsentCollectionLink> keys) {
+        if (keys && !keys.isEmpty()) {
+            List<Issue> issues = Issue.findAllByProjectKeyInList(keys.keySet().toList())
+            issues.each { issue ->
+                // set all documents related to the specified consent group including its consent collection link documents
+                issue.setAttachments(queryService.getAttachmentsForProjects(keys.keySet()))
+                issue.setStatus(keys.get(issue.projectKey).status)
```

Comment:
I'm confused by this ... why are we changing the status of query results? Shouldn't the status be what the status is from the database? As a developer or a user, this would be terribly confusing in the case that these were different. I would see one status in one view and a different status in a different view.

Reply:
RE:
> Status is consent collection link status not issue

The method in question here is not a UI method, it's an underlying query method. We shouldn't be manipulating the underlying data like this in a query method. If the UI needs to display link status, then the UI should be querying for the links and showing that in the right place.